### PR TITLE
Roundstart bugfixes

### LINKED ISSUE #33076: Jukebox music will not play
(WEB REPORT BY: gurfan REMOTE: 206.221.180.138:7777)
# Revision
b01e2822aa68e4d50b7710bf945d83bd90397d5f

music cannot be heard, multiple people are having the same issue

### LINKED ISSUE #33079: People are naked for a significant amount of time at roundstart
Hey hey please fix this. [byond] [system]

### LINKED ISSUE #33086: Spawned as assistant before rolling Nuclear Operative
(WEB REPORT BY: soyconsumer REMOTE: 206.221.180.138:7777)
# Revision
56dcf439d0a85c6a1804faf0797a7e57d0fc27eb

# Description
I spawned in as an assistant in the tool storage while the server was starting the round and everyone was naked. When the server finished loading I got teleported into the Nuke Ops base as a Nukie.

### PR DESCRIPTION
**Pull request by adacovsk**
[hotfix]

Now that the lights start immediately on, turning off empty departments is less of a priority, so this PR moves it to a spot that should affect players less.

## What this does
- Moves some code to a lesser priority function
- Removes `spawn()` from `EquipCustomItems` so everything should be loaded before players are injected
- SetupHooks are actually used to my surprise

## Why it's good
- Closes #33079
- Closes #33076
- Closes #33086
- (maybe, I can't think of any way to fix this right now)

### CODE REVIEW

**Comment by adacovsk** [comment]
@ShiftyRail You implemented the spawn()... Do you think it's okay to remove it in this case? I think it may be causing plasmamen to spawn without suits @WindowsErrors

**Comment by WindowsErrors** [comment]
i ded pleas fix

**Comment by adacovsk** [comment]
> i ded pleas fix

There were two other plasmamen that didn't have this issue, I have no idea how you were singled out, but it's funny

**Comment by WindowsErrors** [comment]
After having thought about it over 6 servings of macaroni and cheese with Spam, I think it might have happened because the round start commonly freezes when I don't clear the cache and the round start froze for a second or two, so I go monkey brain and click reconnect from the client tab out of instinct like I've normally done the past 300 times this has happened, but literally as I clicked reconnect I saw the round start text flash and the screen go black like my character was about to spawn in, but weirdly enough it never felt like I was disconnected, so for multiple seconds I was just the default bald naked guy, long enough for the miners to zip down to my spawn room and ask for the coins like a Team Fortress 2 Scout asking me for the sandvich as they usually do, but this time in perfect position to witness the unique form of love and respect that the /vg/station13 community shows to their code contributors

Basically I clicked reconnect as the round was starting and I think something must have happened in the window where my character had no ckey associated with it

**Comment by adacovsk** [comment]
> After having thought about it over 6 servings of macaroni ... 

That sounds like the issue, thanks that's very helpful. I'll see if I can do something about that in particular

**Comment by ShiftyRail** [comment]
You think? Did you test it?

Anyway that `spawn()` did not have to do with lights but with the DB connection and its very likely that your fix doesn't address it - you just simply dont test it properly locally.

**Comment by adacovsk** [comment]
> You think? Did you test it?
> 
> Anyway that `spawn()` did not have to do with lights but with the DB connection and its very likely that your fix doesn't address it - you just simply dont test it properly locally.

we're not talking about lights, just that now since roundstart is sped up, the DB takes up most of the time during roundstart... My theory is that since there's a spawn when loading custom sprites/items, there can be several seconds where sprites are not loaded. I think it was only a problem with 40+ players

**Comment by ShiftyRail** [comment]
I had a brainfart
but anyway if you remove this `spawn()` you'll just have a delayed roundstart by a few seconds

**Comment by adacovsk** [comment]
> I had a brainfart but anyway if you remove this `spawn()` you'll just have a delayed roundstart by a few seconds

People are freaking out about being naked so delaying it might be necessary...

**Comment by ShiftyRail** [comment]
are they naked gameplay wise or just visually

**Comment by adacovsk** [comment]
> are they naked gameplay wise or just visually

So everyone starts off as your typical admin dummy looking guy and can move around for several seconds, interact, etc. There was one instance that WindowsErrors mentioned where he logged off during this time, and it left him as a naked plasmaman. It's not just visual but I don't think it matters

**Comment by WindowsErrors** [comment]
please bro it's starting to get out of hand

**Comment by adacovsk** [comment]
> please bro it's starting to get out of hand

were you a naked plasmaman again?

**Comment by WindowsErrors** [comment]
maybe

it's not just me either
